packages/camera/camera_android_camerax/android/src/main/java/io/flutter/plugins/camerax/DeviceOrientationManager.java

```diff
@@ -182,7 +183,18 @@ PlatformChannel.DeviceOrientation getUiOrientation() {
    *     Surface.ROTATION_270}
    */
   int getDefaultRotation() {
-    return getDisplay().getRotation();
+    Display display = getDisplay();
+
+    if (display == null) {
+      // The Activity is not available (null, finishing, or destroyed), which can happen briefly
+      // during configuration changes. Returning ROTATION_0 ensures safe fallback and prevents crashes
+      // until a valid Activity is attached again.
+      Log.w(
+          "DeviceOrientationManager", "Cannot get display. Activity may be null during rotation.");
```

Comment:
Nit: can you add to the note something that indicates what you said--that this may be due to the fact that a valid Activity might not be attached just yet due to a race condition?